streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -554,6 +563,58 @@ public void foreach(final ForeachAction<? super K, ? super V> action,
         );
     }
 
+    @Override
+    public KStream<K, V> repartition() {
+        return doRepartition(Repartitioned.as(null));
+    }
+
+    @Override
+    public KStream<K, V> repartition(final Repartitioned<K, V> repartitioned) {
+        return doRepartition(repartitioned);
+    }
+
+    private <KR> KStream<KR, V> doRepartition(final Repartitioned<KR, V> repartitioned) {
```

Comment:
Why do we introduce a new type `<KR>`? The key type of the input and output KStream does not change during repartitioning.